lib/multicell_location/Kconfig

```diff
@@ -20,7 +20,7 @@ if MULTICELL_LOCATION
 
 config MULTICELL_LOCATION_SERVICE_NRF_CLOUD
 	bool "nRF Cloud location service"
-	depends on MODEM_JWT
+	depends on MODEM_JWT || NRF_CLOUD_MQTT
```

Comment:
When looking this now, I'd would be nice if a condition in this level of stack could be like following:
depends on NRF_CLOUD_REST || NRF_CLOUD_MQTT

-> meaning that MODEM_JWT dependency should be handled with in NRF_CLOUD_REST. But in this case, call to nrf_cloud_jwt_generate() should be hidden within nrf_cloud_rest_cell_pos_get(). I.e. would require changes to nrf_cloud api for jwt.

Reply:
There seems to be also option `NRF_CLOUD_REST_AUTOGEN_JWT` which is disabled by default, does it do just that? If it does, that would have to be selected when `NRF_CLOUD_REST` is enabled. There's also `NRF_CLOUD_REST_AUTOGEN_JWT_VALID_TIME_S` to set the token validity time.

Reply:
I think this could be `depends on NRF_CLOUD_REST || NRF_CLOUD_MQTT` with new line added for the autogen JWT like this: `select MODEM_JWT if NRF_CLOUD_REST`

@glarsennordic, what do you think? If this sounds ok, maybe you could do the changes in this PR? Of course the implementation would need to changed a bit.

Reply:
Hah, as it happens, I was the author of the `NRF_CLOUD_REST_AUTOGEN_JWT` feature

It should be relatively simple to remove the JWT token generation from the multicell_location lib and use NRF_CLOUD_REST_AUTOGEN_JWT

The select (I think you meant to write `select NRF_CLOUD_REST_AUTOGEN_JWT  if NRF_CLOUD_REST`?) gets a bit hairy, though -- @jayteemo I think you mentioned at one point feeling we should not use `select` to enable `NRF_CLOUD_REST_AUTOGEN_JWT` for certain libraries. Do you think that applies here?

Reply:
if the location library is updated to use the `NRF_CLOUD_REST_AUTOGEN_JWT` feature, then a `select` seems fine.  

this PR was intended to fix an issue so you can proceed with another PR... so it's up to you if you want to expand the scope by adding new work to it.

Reply:
Awesome! Then I'm on board for doing it @tokangas

The work is pretty much as easy to apply as the original fix, it just means a bit more testing, so it's no problem

Reply:
I have pushed these changes. They are untested, ATM. I will test them today